### ISSUE TITLE
tour: better explanation about type declarations?

### ISSUE THREAD

**Issue by mdempsky** [golang/go#13530]
Context: https://tour.golang.org/moretypes/2

I just decided to look through the Go tour to see how it explains named types and assignability.  As far as I can tell, the only explanation is the parenthetical sentence "And a `type` declaration does what you'd expect."

Based on user questions like issue #13529, I suspect that `type` declarations do *not* necessarily do what users expect, and perhaps it's worth elaborating on this point somehow.

**Comment by MatejLach** [comment]
Yeah, I would word it as "A `type` declaration creates a new, named type, which behaves independently of  its underlying type and thus cannot be used in expressions involving the underlying type without an explicit conversion. "
What do you think @mdempsky ?

**Comment by mjbrender** [comment]
Joining this conversation from #14152. I rewrote it as: 

"A type determines the set of values and operations specific to values of that type. Each declared type has an underlying type that goes back to the [basic types discussed earlier](https://tour.golang.org/basics/11). 

Let's define a type that is also a struct, which is made up of multiple fields."

No matter the wording, I'm a fan of 2-3 sentences here. Types turning into types sounds like a Dr Seuss riddle at first. Smaller, simpler sentences help break that down.

**Comment by zombiezen** [comment]
@broady This is a dupe of golang/tour#121

**Comment by MatejLach** [comment]
@zombiezen If anything, [golang/tour#121](https://github.com/golang/tour/issues/121) is a dupe of thos, given this was reported almost a year prior to the other issue, but I agree that the repo where it is reported is probably better suited for this.
